### PR TITLE
ci: unassign issues stale 14 days ago

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

This closes https://github.com/GreptimeTeam/greptimedb/issues/3525.

## What's changed and what's your intention?

Perhaps we can add more detailed conditions. Welcome to leave a comment :D

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by tisonkun** [comment]
image

This is a demostration reply.

File: .github/workflows/unassign.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Auto Unassign
+on:
+  schedule:
+    - cron: '4 2 * * *'
```
Comment (tisonkun) [inline .github/workflows/unassign.yml:4]: Run every day.

File: .github/workflows/unassign.yml
Comment (tisonkun) [inline .github/workflows/unassign.yml:1]: Current stale issues in our repo:

```
[GreptimeTeam/greptimedb] Open issues updated before 2024-03-03 18:51:48.672521+08:00 (14 days ago) will be considered stale.
[GreptimeTeam/greptimedb] Issue 203 is stale, last updated at 2023-02-14 07:55:13+00:00. Assignees: {'v0y4g3r'}
[GreptimeTeam/greptimedb] Issue 665 is stale, last updated at 2023-03-06 14:41:04+00:00. Assignees: {'ShenJunkun'}
[GreptimeTeam/greptimedb] Issue 1232 is stale, last updated at 2023-03-24 07:51:17+00:00. Assignees: {'fengjiachun'}
[GreptimeTeam/greptimedb] Issue 1023 is stale, last updated at 2023-04-14 12:33:07+00:00. Assignees: {'v0y4g3r'}
[GreptimeTeam/greptimedb] Issue 1412 is stale, last updated at 2023-04-18 12:21:40+00:00. Assignees: {'shuiyisong'}
[GreptimeTeam/greptimedb] Issue 1429 is stale, last updated at 2023-04-20 07:37:49+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 1550 is stale, last updated at 2023-05-16 14:00:07+00:00. Assignees: {'etolbakov'}
[GreptimeTeam/greptimedb] Issue 1033 is stale, last updated at 2023-05-31 11:29:02+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 1672 is stale, last updated at 2023-06-02 10:56:37+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 1819 is stale, last updated at 2023-06-24 14:32:58+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 1790 is stale, last updated at 2023-07-03 11:51:40+00:00. Assignees: {'ccjeff'}
[GreptimeTeam/greptimedb] Issue 1496 is stale, last updated at 2023-07-10 03:30:30+00:00. Assignees: {'evenyag'}
[GreptimeTeam/greptimedb] Issue 1988 is stale, last updated at 2023-07-18 09:12:11+00:00. Assignees: {'v0y4g3r'}
[GreptimeTeam/greptimedb] Issue 1915 is stale, last updated at 2023-07-24 06:39:15+00:00. Assignees: {'etolbakov'}
[GreptimeTeam/greptimedb] Issue 1606 is stale, last updated at 2023-07-24 07:23:24+00:00. Assignees: {'fengys1996'}
[GreptimeTeam/greptimedb] Issue 2031 is stale, last updated at 2023-07-25 07:53:35+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 1662 is stale, last updated at 2023-08-09 06:47:03+00:00. Assignees: {'Taylor-lagrange'}
[GreptimeTeam/greptimedb] Issue 1995 is stale, last updated at 2023-08-18 10:09:05+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 1310 is stale, last updated at 2023-08-21 11:33:20+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 2013 is stale, last updated at 2023-08-22 07:52:19+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 1168 is stale, last updated at 2023-08-22 09:40:19+00:00. Assignees: {'Gump9'}
[GreptimeTeam/greptimedb] Issue 1378 is stale, last updated at 2023-09-01 02:29:24+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 412 is stale, last updated at 2023-09-01 02:36:09+00:00. Assignees: {'v0y4g3r', 'waynexia'}
[GreptimeTeam/greptimedb] Issue 1823 is stale, last updated at 2023-09-01 02:50:49+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 1709 is stale, last updated at 2023-09-14 02:02:32+00:00. Assignees: {'NiwakaDev'}
[GreptimeTeam/greptimedb] Issue 996 is stale, last updated at 2023-09-17 07:13:29+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 2220 is stale, last updated at 2023-09-26 05:49:55+00:00. Assignees: {'TheWaWaR'}
[GreptimeTeam/greptimedb] Issue 336 is stale, last updated at 2023-10-11 03:01:42+00:00. Assignees: {'evenyag', 'waynexia'}
[GreptimeTeam/greptimedb] Issue 602 is stale, last updated at 2023-10-17 09:07:50+00:00. Assignees: {'Lilit0x'}
[GreptimeTeam/greptimedb] Issue 1906 is stale, last updated at 2023-10-17 09:11:04+00:00. Assignees: {'Taylor-lagrange'}
[GreptimeTeam/greptimedb] Issue 2505 is stale, last updated at 2023-10-24 13:05:51+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 2596 is stale, last updated at 2023-10-30 06:38:04+00:00. Assignees: {'Lilit0x'}
[GreptimeTeam/greptimedb] Issue 2702 is stale, last updated at 2023-11-07 06:18:06+00:00. Assignees: {'evenyag'}
[GreptimeTeam/greptimedb] Issue 1267 is stale, last updated at 2023-11-13 12:07:12+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 2618 is stale, last updated at 2023-11-14 15:53:58+00:00. Assignees: {'realtaobo'}
[GreptimeTeam/greptimedb] Issue 2798 is stale, last updated at 2023-11-23 09:05:26+00:00. Assignees: {'evenyag', 'waynexia'}
[GreptimeTeam/greptimedb] Issue 2902 is stale, last updated at 2023-12-11 09:32:18+00:00. Assignees: {'WenyXu'}
[GreptimeTeam/greptimedb] Issue 2922 is stale, last updated at 2023-12-13 03:51:59+00:00. Assignees: {'yuanbohan'}
[GreptimeTeam/greptimedb] Issue 2805 is stale, last updated at 2023-12-20 02:51:31+00:00. Assignees: {'evenyag', 'QuenKar'}
[GreptimeTeam/greptimedb] Issue 2816 is stale, last updated at 2023-12-28 03:56:17+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 3063 is stale, last updated at 2024-01-02 03:09:29+00:00. Assignees: {'discord9'}
[GreptimeTeam/greptimedb] Issue 2663 is stale, last updated at 2024-01-02 03:32:56+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 2369 is stale, last updated at 2024-01-02 03:39:43+00:00. Assignees: {'WenyXu'}
[GreptimeTeam/greptimedb] Issue 2101 is stale, last updated at 2024-01-02 03:43:53+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 1719 is stale, last updated at 2024-01-02 03:47:22+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 3004 is stale, last updated at 2024-01-02 05:03:15+00:00. Assignees: {'CrystalAnalyst'}
[GreptimeTeam/greptimedb] Issue 3070 is stale, last updated at 2024-01-02 08:14:21+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 1996 is stale, last updated at 2024-01-02 15:24:00+00:00. Assignees: {'WenyXu'}
[GreptimeTeam/greptimedb] Issue 2532 is stale, last updated at 2024-01-04 08:32:34+00:00. Assignees: {'killme2008'}
[GreptimeTeam/greptimedb] Issue 2657 is stale, last updated at 2024-01-04 22:52:28+00:00. Assignees: {'Abhishek-Mallick'}
[GreptimeTeam/greptimedb] Issue 3104 is stale, last updated at 2024-01-05 10:18:39+00:00. Assignees: {'evenyag'}
[GreptimeTeam/greptimedb] Issue 3072 is stale, last updated at 2024-01-07 21:09:52+00:00. Assignees: {'QuenKar'}
[GreptimeTeam/greptimedb] Issue 3074 is stale, last updated at 2024-01-09 06:57:00+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 3153 is stale, last updated at 2024-01-12 04:02:52+00:00. Assignees: {'MichaelScofield'}
[GreptimeTeam/greptimedb] Issue 2705 is stale, last updated at 2024-01-12 12:45:28+00:00. Assignees: {'zhongzc'}
[GreptimeTeam/greptimedb] Issue 3172 is stale, last updated at 2024-01-16 06:49:10+00:00. Assignees: {'WenyXu'}
[GreptimeTeam/greptimedb] Issue 1042 is stale, last updated at 2024-01-22 08:23:36+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 2965 is stale, last updated at 2024-01-23 10:30:21+00:00. Assignees: {'evenyag'}
[GreptimeTeam/greptimedb] Issue 3221 is stale, last updated at 2024-01-23 13:56:31+00:00. Assignees: {'evenyag'}
[GreptimeTeam/greptimedb] Issue 3230 is stale, last updated at 2024-01-24 13:36:35+00:00. Assignees: {'QuenKar'}
[GreptimeTeam/greptimedb] Issue 3235 is stale, last updated at 2024-01-25 03:39:18+00:00. Assignees: {'QuenKar'}
[GreptimeTeam/greptimedb] Issue 3233 is stale, last updated at 2024-02-02 01:12:05+00:00. Assignees: {'WenyXu'}
[GreptimeTeam/greptimedb] Issue 3044 is stale, last updated at 2024-02-02 10:45:01+00:00. Assignees: {'KushnerykPavel'}
[GreptimeTeam/greptimedb] Issue 3045 is stale, last updated at 2024-02-07 05:32:35+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 3294 is stale, last updated at 2024-02-07 05:42:12+00:00. Assignees: {'fengjiachun'}
[GreptimeTeam/greptimedb] Issue 3041 is stale, last updated at 2024-02-07 06:00:04+00:00. Assignees: {'niebayes'}
[GreptimeTeam/greptimedb] Issue 1830 is stale, last updated at 2024-02-28 09:39:27+00:00. Assignees: {'v0y4g3r'}
[GreptimeTeam/greptimedb] Issue 2579 is stale, last updated at 2024-02-28 09:39:55+00:00. Assignees: {'zyy17'}
[GreptimeTeam/greptimedb] Issue 2583 is stale, last updated at 2024-02-28 09:40:23+00:00. Assignees: {'zyy17'}
[GreptimeTeam/greptimedb] Issue 2948 is stale, last updated at 2024-02-28 09:43:08+00:00. Assignees: {'waynexia'}
[GreptimeTeam/greptimedb] Issue 2865 is stale, last updated at 2024-02-28 09:55:15+00:00. Assignees: {'fengjiachun'}
[GreptimeTeam/greptimedb] Issue 638 is stale, last updated at 2024-02-28 10:01:39+00:00. Assignees: {'discord9'}
[GreptimeTeam/greptimedb] Issue 3408 is stale, last updated at 2024-02-28 13:36:57+00:00. Assignees: {'WenyXu'}
```

**Comment by sunng87** [comment]
May I know the criteria for an issue being inactive?

**Comment by tisonkun** [comment]
@sunng87 currently it checks the "updated_at" of the issue property.

The logic is [here](https://github.com/tisonspieces/auto-unassign/blob/d02228d39dd1ca000a57c7bc0e6826b016d568dc/src/auto_unassign/__init__.py#L46).

I created https://github.com/tisonspieces/auto-unassign/issues/2 for any further conditions we'd like to refine.

**Review by MichaelScofield — APPROVED** [review]

**Review by sunng87 — APPROVED** [review]

**Pull request merged**